File: workspace/scripts/check-deposit-streak-ui.mjs
import { chromium } from 'playwright';
import { fileURLToPath } from 'url';
import path from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function checkDepositStreakUI() {
  console.log('🚀 Starting Deposit Streak UI Check...');

  const browser = await chromium.launch({
    headless: false,
    args: ['--start-maximized']
  });

  const context = await browser.newContext({
    viewport: { width: 1920, height: 1080 },
    locale: 'en-US'
  });

  const page = await context.newPage();

  const email = 'demo1772643519424@nextcode.tech';
  const password = 'Qweasd123!';
  const timestamp = new Date().toISOString().replace(/[:.]/g, '-');
  const screenshotDir = path.join(__dirname, '..', 'screenshots', 'deposit-streak-check');

  console.log(`📸 Screenshots will be saved to: ${screenshotDir}`);

  // Navigate to Minebit PROD
  console.log('🌐 Navigating to Minebit PROD...');
  await page.goto('https://minebit-casino.prod.sofon.one', { waitUntil: 'networkidle' });
  await page.screenshot({ path: path.join(screenshotDir, `01-home-${timestamp}.png`), fullPage: true });
  console.log('✅ Screenshot: Homepage saved');

  // Find and click login button
  console.log('🔍 Looking for login button...');
  try {
    // Try multiple possible login button selectors
    const loginSelectors = [
      'button:has-text("Log in")',
      'button:has-text("Login")',
      'a[href*="login"]',
      '[data-testid*="login"]',
      '.login-button',
      '.btn-login'
    ];

    let loginButton = null;
    for (const selector of loginSelectors) {
      try {
        loginButton = await page.waitForSelector(selector, { timeout: 2000 });
        if (loginButton) {
          console.log(`✅ Found login button with selector: ${selector}`);
          await loginButton.click();
          break;
        }
      } catch (e) {
        continue;
      }
    }

    if (!loginButton) {
      console.log('⚠️ Could not find login button with standard selectors');
      console.log('🔍 Page title:', await page.title());
      throw new Error('Login button not found');
    }

    await page.waitForLoadState('networkidle');
    console.log('✅ Login button clicked');

  } catch (e) {
    console.log('❌ Error clicking login button:', e.message);
    await page.screenshot({ path: path.join(screenshotDir, `error-login-${timestamp}.png`), fullPage: true });
    await browser.close();
    return;
  }

  // Fill login form
  console.log('📝 Filling login form...');
  try {
    // Find email input
    const emailSelectors = [
      'input[type="email"]',
      'input[name*="email"]',
      'input[placeholder*="email" i]',
      'input[placeholder*="Email" i]'
    ];

    let emailInput = null;
    for (const selector of emailSelectors) {
      try {
        emailInput = await page.waitForSelector(selector, { timeout: 2000 });
        if (emailInput) {
          console.log(`✅ Found email input with selector: ${selector}`);
          break;
        }
      } catch (e) {
        continue;
      }
    }

    if (!emailInput) {
      throw new Error('Email input not found');
    }

    await emailInput.fill(email);
    console.log(`✅ Email filled: ${email}`);

    // Find password input
    const passwordSelectors = [
      'input[type="password"]',
      'input[name*="password"]',
      'input[placeholder*="password" i]',
      'input[placeholder*="Password" i]'
    ];

    let passwordInput = null;
    for (const selector of passwordSelectors) {
      try {
        passwordInput = await page.waitForSelector(selector, { timeout: 2000 });
        if (passwordInput) {
          console.log(`✅ Found password input with selector: ${selector}`);
          break;
        }
      } catch (e) {
        continue;
      }
    }

    if (!passwordInput) {
      throw new Error('Password input not found');
    }

    await passwordInput.fill(password);
    console.log('✅ Password filled');

    // Find and click submit button
    const submitSelectors = [
      'button[type="submit"]',
      'button:has-text("Log in")',
      'button:has-text("Login")',
      'button:has-text("Sign in")',
      '[type="submit"]'
    ];

    let submitButton = null;
    for (const selector of submitSelectors) {
      try {
        submitButton = await page.waitForSelector(selector, { timeout: 2000 });
        if (submitButton) {
          console.log(`✅ Found submit button with selector: ${selector}`);
          break;
        }
      } catch (e) {
        continue;
      }
    }

    if (!submitButton) {
      throw new Error('Submit button not found');
    }

    await submitButton.click();
    console.log('✅ Login form submitted');

  } catch (e) {
    console.log('❌ Error filling login form:', e.message);
    await page.screenshot({ path: path.join(screenshotDir, `error-form-${timestamp}.png`), fullPage: true });
    await browser.close();
    return;
  }

  // Wait for login to complete
  console.log('⏳ Waiting for login to complete...');
  await page.waitForLoadState('networkidle');
  await page.waitForTimeout(3000);

  // Screenshot after login
  await page.screenshot({ path: path.join(screenshotDir, `02-logged-in-${timestamp}.png`), fullPage: true });
  console.log('📸 Screenshot: Logged in (full page)');

  // Navigate to bonuses page
  console.log('🎁 Navigating to bonuses page...');
  try {
    await page.goto('https://minebit-casino.prod.sofon.one/bonuses', { waitUntil: 'networkidle' });
    console.log('✅ Navigated to bonuses page');
  } catch (e) {
    console.log('⚠️ Could not navigate to bonuses page:', e.message);
  }

  // Wait for bonuses to load
  await page.waitForTimeout(2000);

  // Screenshot bonuses page
  await page.screenshot({ path: path.join(screenshotDir, `03-bonuses-${timestamp}.png`), fullPage: true });
  console.log('📸 Screenshot: Bonuses page (full page)');

  // Check for Deposit Streak bonus
  console.log('\n🔍 Checking for Deposit Streak bonus...');
  try {
    const depositStreakExists = await page.locator(':has-text("Deposit Streak")').count() > 0 ||
                                await page.locator(':has-text("deposit streak")').count() > 0 ||
                                await page.locator(':has-text("Streak")').count() > 0;

    if (depositStreakExists) {
      console.log('✅ DEPOSIT STREAK BONUS FOUND!');

      const streakElement = page.locator(':has-text("Streak")').first();
      await streakElement.screenshot({ path: path.join(screenshotDir, `04-deposit-streak-card-${timestamp}.png`) });
      console.log('📸 Screenshot: Deposit Streak bonus element saved');
    } else {
      console.log('❌ DEPOSIT STREAK BONUS NOT FOUND');
    }
  } catch (e) {
    console.log('⚠️ Error checking for Deposit Streak:', e.message);
  }

  // Check for any active bonuses
  console.log('\n🔍 Checking for active bonuses...');
  try {
    const claimButtons = await page.locator('button:has-text("Claim")').count();
    console.log(`🎁 Found ${claimButtons} "Claim" buttons`);

    const activeTexts = await page.locator(':has-text("Active"), :has-text("Available")').count();
    console.log(`🎁 Found ${activeTexts} elements with "Active" or "Available" text`);

    // Count all bonus cards
    const bonusCards = await page.locator('[class*="bonus"], [data-testid*="bonus"], .bonus-card').count();
    console.log(`🎁 Total bonus-related elements: ${bonusCards}`);

  } catch (e) {
    console.log('⚠️ Error checking for bonuses:', e.message);
  }

  // Output summary
  console.log('\n📊 SUMMARY:');
  console.log('   - Screenshots saved to:', screenshotDir);
  console.log('   - Test user:', email);
  console.log('   - Client ID: 1181008');
  console.log('   - Deposits made: 10 × $30 USD = $300');

  // Keep browser open for manual inspection
  console.log('\n⏸️ Browser will stay open for 60 seconds for manual inspection...');
  await page.waitForTimeout(60000);

  await browser.close();
  console.log('\n✅ Done! Browser closed.');
}

checkDepositStreakUI().catch(console.error);

File: workspace/scripts/check_deposit_streak_ui.py
#!/usr/bin/env python3
"""
Manual QA: Check Deposit Streak bonuses on PROD UI after Python script deposits.
Uses Playwright to login, navigate to bonuses page, and take screenshots.
"""

import asyncio
from playwright.async_api import async_playwright
from datetime import datetime

async def check_bonuses_ui():
    """
    Login to Minebit PROD, navigate to bonuses page, take screenshots.
    """
    async with async_playwright() as p:
        # Launch browser
        browser = await p.chromium.launch(headless=False)
        context = await browser.new_context(
            viewport={'width': 1920, 'height': 1080},
            locale='en-US',
            timezone_id='Europe/Vienna'
        )

        # Test user credentials
        email = "demo1772643519424@nextcode.tech"
        password = "Qweasd123!"

        page = await context.new_page()

        # Navigate to login page
        await page.goto("https://minebit-casino.prod.sofon.one", wait_until="networkidle")
        await page.screenshot(path=f"/tmp/minebit_home_{datetime.now().strftime('%H%M%S')}.png")

        print("📸 Screenshot: Homepage saved")

        # Find and click login button
        try:
            # Try multiple selectors for login button
            login_button = await page.wait_for_selector(
                "button:has-text('Log in'), button:has-text('Login'), [data-testid*='login'], a[href*='login']",
                timeout=5000
            )
            await login_button.click()
            print("✅ Login button clicked")
        except Exception as e:
            print(f"⚠️ Could not find login button: {e}")
            print("🔍 Page content (first 500 chars):")
            print(await page.content()[:500])
            await browser.close()
            return

        # Wait for login form
        await page.wait_for_load_state("networkidle")

        # Fill login form
        try:
            email_input = await page.wait_for_selector("input[type='email'], input[name*='email'], input[placeholder*='email']", timeout=5000)
            await email_input.fill(email)
            print(f"✅ Email filled: {email}")

            password_input = await page.wait_for_selector("input[type='password'], input[name*='password'], input[placeholder*='password']", timeout=5000)
            await password_input.fill(password)
            print("✅ Password filled")

            # Submit form
            submit_button = await page.wait_for_selector("button[type='submit'], button:has-text('Log in'), button:has-text('Login')", timeout=5000)
            await submit_button.click()
            print("✅ Login form submitted")

        except Exception as e:
            print(f"❌ Could not fill login form: {e}")
            await page.screenshot(path=f"/tmp/login_form_error_{datetime.now().strftime('%H%M%S')}.png")
            await browser.close()
            return

        # Wait for login to complete
        await page.wait_for_load_state("networkidle")
        await asyncio.sleep(2)

        # Take screenshot after login
        await page.screenshot(path=f"/tmp/minebit_logged_in_{datetime.now().strftime('%H%M%S')}.png", full_page=True)
        print("📸 Screenshot: Logged in page saved (full page)")

        # Navigate to bonuses page
        try:
            # Try to find bonuses link/button
            bonuses_link = await page.wait_for_selector(
                "a:has-text('Bonus'), a:has-text('Bonuses'), [data-testid*='bonus'], nav a[href*='bonus']",
                timeout=10000
            )
            await bonuses_link.click()
            print("✅ Navigated to bonuses page")
        except Exception as e:
            print(f"⚠️ Could not find bonuses link: {e}")
            print("🔍 Trying direct URL...")
            await page.goto("https://minebit-casino.prod.sofon.one/bonuses", wait_until="networkidle")

        # Wait for bonuses to load
        await asyncio.sleep(2)

        # Take screenshot of bonuses page
        await page.screenshot(path=f"/tmp/minebit_bonuses_{datetime.now().strftime('%H%M%S')}.png", full_page=True)
        print("📸 Screenshot: Bonuses page saved (full page)")

        # Check for bonus cards
        bonus_cards = await page.locator("[data-testid*='bonus'], .bonus-card, [class*='bonus']").all()
        print(f"\n🎁 Found {len(bonus_cards)} bonus elements on page")

        # Try to find Deposit Streak bonus
        try:
            deposit_streak = await page.wait_for_selector(":has-text('Deposit Streak'), :has-text('deposit streak'), :has-text('streak')", timeout=5000)
            if deposit_streak:
                print("✅ Deposit Streak bonus found on page!")
                # Screenshot the specific element
                await deposit_streak.screenshot(path=f"/tmp/deposit_streak_bonus_{datetime.now().strftime('%H%M%S')}.png")
                print("📸 Screenshot: Deposit Streak bonus element saved")
        except:
            print("❌ Deposit Streak bonus NOT found on page")

        # Check for any active bonuses
        print("\n🔍 Checking for active bonuses...")
        try:
            active_bonuses = await page.locator(":has-text('Claim'), :has-text('Active'), :has-text('Available')").all()
            print(f"   Found {len(active_bonuses)} potentially active bonuses")
        except:
            print("   No active bonuses found")

        # Keep browser open for manual inspection
        print("\n⏸️ Browser stays open for manual inspection. Press Ctrl+C to close.")
        try:
            await asyncio.sleep(60)  # Keep open for 60 seconds
        except KeyboardInterrupt:
            print("\n👋 Closing browser...")

        await browser.close()

if __name__ == "__main__":
    asyncio.run(check_bonuses_ui())
